modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/MinimalApplicationModelGenerator.java

```diff
@@ -0,0 +1,162 @@
+ * {@link ComponentModel}s that must be created for the requested {@link ComponentModel} to
+ * work properly.
+ *
+ * @since 4.0
+ */
+public class MinimalApplicationModelGenerator {
+
+  private final ComponentBuildingDefinitionRegistry componentBuildingDefinitionRegistry;
+  private ApplicationModel applicationModel;
+
+  /**
+   * Creates a new instance associated to a complete {@link ApplicationModel}.
+   *
+   * @param applicationModel the artifact {@link ApplicationModel}.
+   * @param componentBuildingDefinitionRegistry the registry to find the {@link ComponentBuildingDefinition}s  associated to each {@link ComponentModel} that must be resolved.
+     */
+  public MinimalApplicationModelGenerator(ApplicationModel applicationModel,
+                                          ComponentBuildingDefinitionRegistry componentBuildingDefinitionRegistry) {
+    this.applicationModel = applicationModel;
+    this.componentBuildingDefinitionRegistry = componentBuildingDefinitionRegistry;
+  }
+
+  /**
+   * Resolves the minimal set of {@link ComponentModel}s for a message processor
+   * within a flow.
+   *
+   * @param processorPath the processor path in which the message processor is located.
+   * @return the generated {@link ApplicationModel} with the minimal set of {@link ComponentModel}s required.
+   * @throws NoSuchComponentModelException if the requested component does not exists.
+   */
+  public ApplicationModel getMinimalModelByPath(String processorPath) {
+    String[] parts = processorPath.split("/");
+    String flowName = parts[0];
+    ComponentModel flowModel = findRequiredComponentModel(flowName);
+    filterFlowModelParts(flowModel, parts);
+    return getMinimalModelByName(flowModel.getNameAttribute());
+  }
+
+  /**
+   * Resolves the minimal set of {@link ComponentModel}s for a named component
+   * within the configuration.
+   *
+   * @param name name of the {@link ComponentModel}
+   * @return the generated {@link ApplicationModel} with the minimal set of {@link ComponentModel}s required.
+   * @throws NoSuchComponentModelException if the requested component does not exists.
+   */
+  public ApplicationModel getMinimalModelByName(String name) {
+    ComponentModel requestedComponentModel = findRequiredComponentModel(name);
+    final Set<String> otherRequiredGlobalComponents = resolveComponentDependencies(requestedComponentModel);
+    otherRequiredGlobalComponents.add(name);
+    Set<String> allRequiredComponentModels = findComponentModelsDependencies(otherRequiredGlobalComponents);
+    Iterator<ComponentModel> iterator = applicationModel.getRootComponentModel().getInnerComponents().iterator();
+    while (iterator.hasNext()) {
+      ComponentModel componentModel = iterator.next();
+      if (componentModel.getNameAttribute() == null || !allRequiredComponentModels.contains(componentModel.getNameAttribute())) {
+        iterator.remove();
```

Comment:
For now yes, as part of the remaining task for DSL processing is to refactor this code to make it immutable. But I will tackle that later. Hopefully it can be done before 4.0 but since it's not part of the API I can do that whole refactor in a minor version.